src/components/questions-accordion/QuestionsAccordion.tsx

```diff
@@ -0,0 +1,93 @@
+import { FC, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import Typography from '@mui/material/Typography'
+import Box from '@mui/material/Box'
+import Accordion from '@mui/material/Accordion'
+import AccordionSummary from '@mui/material/AccordionSummary'
+import AccordionDetails from '@mui/material/AccordionDetails'
+
+import ArrowForwardIosSharpIcon from '@mui/icons-material/ArrowForwardIosSharp'
+
+import { AccordionItem } from '~/types'
+import { styles } from '~/components/questions-accordion/QuestionsAccordion.styles'
+
+interface QuestionsAccordion {
+  items: AccordionItem[]
+  title?: string
+  showIcon?: boolean
+}
+
+const QuestionsAccordion: FC<QuestionsAccordion> = ({
+  items,
+  title = '',
+  showIcon = true
+}) => {
+  const [activeQuestions, setActiveQuestions] = useState<number[]>([])
+
+  const onChange = (activeQuestion: number) => {
+    setActiveQuestions((prevActiveQuestions) => {
+      if (prevActiveQuestions.includes(activeQuestion)) {
+        return prevActiveQuestions.filter(
+          (prevActiveQuestion) => prevActiveQuestion !== activeQuestion
+        )
+      } else {
+        return [...prevActiveQuestions, activeQuestion]
+      }
+    })
+  }
+
+  const { t } = useTranslation()
+
+  return (
+    <Box sx={title ? styles.container : {}}>
+      {title ? (
+        <Typography sx={styles.title} variant='h5'>
+          {t(title)}
+        </Typography>
+      ) : null}
+      {items.map((item, index) => (

```

Comment:
you are using here code almost similar to Accordions component, why you couldn't use it here?

Reply:
because Accordions component is not what is need to create this QuestionsAccordion for the next reasons:
- QuestionsAccordion component

Reply:
because Accordions component is not what is need to create this QuestionsAccordion for the next reasons:
- QuestionsAccordion component

Reply:
because Accordions component is not what i needed to create QuestionsAccordion for the next reasons:
- QuestionsAccordion is using logic not the same as Accordions, cause in the first one we can open all items as it is shown in Figma mark up, in the Accordion we can open only and if we open one other will be close automatically
- QuestionsAccordion component is using another icon than Accordions
- Accordions had some props for MUI Accordion component that were no need in QuestionsAccordion so i thought it will be too much props to add to Accordions and it will be cleaner and better to create from MUI Accordion component
- Also some micro moments connected with styling, where i couldn't make looking of existing Accordion component to what i needed for QuestionsAccordion, mostly it was concerned with positioning

Reply:
for me its looking like minor changes, we should not create new component for ewery need we have

Reply:
if accordions component is too different you have to create a more general component which we could reuse, but yours is too specified, and when we will need another accordion we will create one more what is bad

Reply:
i understand that reusing is much better and needs performing by maximum, but i still don't know how combine two different logis of opening accordions, depends on what it should change, do you have any ideas?